sklearn/feature_selection/from_model.py

```diff
@@ -183,10 +187,22 @@ class SelectFromModel(BaseEstimator, SelectorMixin):
     `threshold_`: float
         The threshold value used for feature selection.
     """
-    def __init__(self, estimator, threshold=None, prefit=False):
+    def __init__(self, estimator, threshold=None, prefit=False,
+                 max_n_features=None):
         self.estimator = estimator
         self.threshold = threshold
         self.prefit = prefit
+        self.max_n_features = max_n_features
+
+    def _check_params(self, X, y):
+        X, y = check_X_y(X, y)
+        if self.max_n_features is not None and not (
+                                    self.max_n_features == "all" or
```

Comment:
The "all" option is available for `SelectKBest`. So it makes sense that `SelectFromModel` supports this option.